### PR TITLE
Refactor for v0.5

### PR DESCRIPTION
**Pull request by vchuravy**
@SimonDanisch This is my current status. The last piece missing is the `call` deprecation and some more cleanups.

### CODE REVIEW

**Comment by yuyichao** [comment]
@vchuravy Ping.

**Comment by vchuravy** [comment]
pong. I will polish this today.

File: README.md [hunk 1]
```diff
@@ -55,7 +55,6 @@ OpenCL.jl has had contributions from [many developers](https://github.com/JuliaG
 ```julia
 
 import OpenCL
```
Comment (yuyichao) [inline README.md]: you mean `using`?

**Comment by vchuravy** [comment]
One discussion point would be the future of `cl.call`

File: README.md (hunk 1 above)
Comment (vchuravy) [inline README.md]: thanks for noticing.

**Comment by vchuravy** [comment]
@yuyichao Thanks for looking over this, I am thinking about merging this later. and tagging a new minor version. There is still more refactoring to do, but I don't know when or if I get to it.

**Comment by SimonDanisch** [comment]
Lovely! :)

**Comment by vchuravy** [comment]
@dfdx This might be of interest for you for ClBLAS and CLFFT

**Comment by dfdx** [comment]
@vchuravy Thanks for notifying me. Yes, I already have a bunch of tasks for this weekend, and updating CLBLAS and CLFFT is on the list :)

**Pull request merged**